File: Backend/PlantStore.Api/DTOs/ProductDto.cs
namespace PlantStore.Api.Dtos
{
    public class ProductDto
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal Price { get; set; }

        public List<string> ImageUrls { get; set; } = new();
        public string? DefaultImageUrl { get; set; } // 👈 nowość

        public int InStock { get; set; }
        public int CategoryId { get; set; }
        public string? CategoryName { get; set; }
    }
}

File: Frontend/plant-store-frontend/src/types/ProductDto.ts
export interface ProductDto {
  id: number;
  name: string;
  description: string;
  price: number;
  imageUrls: string[];
  defaultImageUrl?: string;
  inStock: number;
  categoryId: number;
  categoryName: string | null;
}

// DTO do tworzenia i edycji (bez id i categoryName)
export type ProductCreateDto = Omit<ProductDto, "id" | "categoryName">;
export type ProductUpdateDto = ProductCreateDto;
